File: backend/Domain.fs
namespace FamilyEventsBackend.Domain

open System

type Time = { HH: int; MM: int }

type Event =
    { Id: Guid
      Title: string
      StartTime: Time
      EndTime: Time }

type Person = { Id: Guid; NickName: string }

type ScheduleEntry = { Event: Event; Persons: Person list }

type Day =
    { DayOfWeek: DayOfWeek
      Entries: ScheduleEntry list }

type Week = { WeekNumber: int; Days: Day list }

File: backend/Domain.cs
using System;
using System.Collections.Generic;

namespace FamilyEventsBackend.Domain
{
    public record Time(int HH, int MM);

    public record Event(Guid Id, string Title, Time StartTime, Time EndTime);

    public record Person(Guid Id, string NickName);

    public record ScheduleEntry(Event Event, List<Person> Persons);

    public record Day(DayOfWeek DayOfWeek, List<ScheduleEntry> Entries);

    public record Week(int WeekNumber, List<Day> Days);
}